### DISCUSSION TITLE
How does one debug 500: Internal Server Errors on Vercel?

### DISCUSSION THREAD

**Discussion by amityweb** [vercel/next.js#62792] [Help]
### Summary

My website builds fine and works fine when built and all static. But when using Draft Mode, I get a 500: INTERNAL_SERVER_ERROR. 

Vercel logs show the request response as a 200, and there are no errors shown. 

It also works fine on my localhost and no errors. 

I have been working on this site for a LONG time! Last year, and never had this. Just starting happening recently. 

I appreciate this error may be related to my code but how would I find out when there is no error log for it and it works on my dev machine fine? 

This is the specific error, which is not even mentioned on their error code page:

```
500: INTERNAL_SERVER_ERROR
Code: INTERNAL_SERVER_ERROR
ID: lhr1::iad1::877wv-1709491511486-f354b670c2e5
```

Thanks

### Additional information

```js
"next": "^13.4.8" - using pages/_app.js not app folder
```

**Comment by icyJoseph** [comment]
I guess this is happening to a single endpoint? If so, then you could try to make sure all potential exceptions are handled, either try/catch or any equivalent. Maybe a key/token has expired, and there's no handling for such error case.

**Comment by amityweb** [comment]
It's all endpoints.

Wouldn't such issues be picked up during the build? It's all mostly static content and it's all fetched ok at build, and the pages created. There's some dynamic content fetched via api, I haven't changed anything in there.

It's only in draft mode this happens. But then doesn't draft mode just do a rebuild of the page on each request so you can see changes like with content? If so, it's confusing why it works on the main build but then not when in draft mode.

**Reply by icyJoseph** [reply]
Right, are you using the same Node.js in prod and locally?

You didn't upgrade Next.js at all right?

**Reply by amityweb** [reply]
Same node yes and I haven't updated next. I went back many months in GitHub to double check! Same version still. 

Something I've noticed... I can actually have the error logs report a 500 error. Although there's no reason why it happened included. The log seems to happen only some pages (even though it occurs on all pages). But when the 500 log shows, it is immediately followed by a log of the same endpoint returning a 200 response (or a 404 if I enter any old slug). I don't know the inner workings of Vercel but I find that weird.

**Comment by amityweb** [comment]
I decided to ask Vercel to help, I have spent ages on this. But I need a paid account as I was on Hobby for development. As soon as I upgraded and the site was transferred to my paid Pro account it started working. 

I still submitted the ticket, I would like to see what Vercel say about it.

**Reply by icyJoseph** [reply]
Maybe they hardened a limit? Lambda size or timeout time? Weird, please do post back once you know what happened.
